### PR TITLE
[14.0][ADD] product_variant_default_code_constraint module

### PR DESCRIPTION
**Pull request by Kev-Roche**
Extension of the product_variant_default_code module, adding more constraints and restrictions on the differents default_code parts, according to the discussion here : https://github.com/OCA/product-variant/pull/227

### CODE REVIEW

File: product_variant_default_code_constraint/models/product.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2022 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class ProductTemplate(models.Model):
```
Comment (rousseldenis) [inline product_variant_default_code_constraint/models/product.py]: Even if this adds not so many code, should be split in separate files.

File: product_variant_default_code_constraint/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This module add "unique" constraints and flagthe components of variant default_code as required :
+- code (product.attribute)
```
Comment (rousseldenis) [inline product_variant_default_code_constraint/readme/DESCRIPTION.rst]:
```suggestion
* code (product.attribute)
```

Comment (rousseldenis) [reply product_variant_default_code_constraint/readme/DESCRIPTION.rst]: Same for others

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by Kev-Roche** [comment]
@rousseldenis , Thank you for your time review, I applied your improvements.

**Comment by sebastienbeau** [comment]
@Kev-Roche can you rebase ?

**Comment by Kev-Roche** [comment]
replace by : https://github.com/OCA/product-variant/pull/262
Only because of a wrong rebase (my bad)

**Pull request closed without merging**